runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -586,6 +588,15 @@ public void start() {
         options.getWindmillHarnessUpdateReportingPeriod().getMillis(),
         TimeUnit.MILLISECONDS);
 
+    workerMessageReportTimer = executorSupplier.apply("WorkerMessageTimer");
```

Comment:
this needs to be shutdown and awaited in stop() below like the other timers.

You could consider refactoring so that we have a collection of these timers that we add to after we start them, since the shutdown/await code is repetitive and I don't think we otherwise refer to them.

Reply:
Refactored and fixed.